### ISSUE TITLE
Platform-independent types used in the internals

### ISSUE THREAD

**Issue by Ratstail91** [Ratstail91/Toy#35]
Currently, there are size checks for different datatypes within `common.c`, but this isn't ideal. I'd like to change all of the variable types to match the following pattern:

* s32 -> 32 bit signed int
* u16 -> 16 bit unsigned short
* b8 -> 8 bit boolean
* f32 -> 32 bit float
* etc.

These could be based on the types in `stdint.h`

This is low priority, but will need to be done before version 1.0.0.

**Comment by Ratstail91** [comment]
This was not finished by 1.0.0. Instead, I don't think it's essential or needed in any way, shape or form.

**Comment by Ratstail91** [comment]
You're gonna need these for weird platforms, dummy.

**Comment by Ratstail91** [comment]
I'll need to embed a byte-flag to indicate what kind of internal byte format is used, probably in the header.

**Comment by Ratstail91** [comment]
This will be solved in version 2+.
